src/cursor/cur_backup_incr.c

```diff
@@ -128,22 +117,39 @@ __curbackup_incr_next(WT_CURSOR *cursor)
         cb->nbits = 0;
         cb->offset = 0;
         cb->bit_offset = 0;
+        /*
+         * By setting this to true, the next call will detect we're done in the code for the
+         * incremental cursor below and return WT_NOTFOUND.
+         */
         cb->incr_init = true;
         __wt_cursor_set_key(cursor, 0, size, WT_BACKUP_FILE);
     } else {
-        /*
-         * We don't have this object's incremental information, and it's not a full file copy. Get a
-         * list of the block modifications for the file. The block modifications are from the
-         * incremental identifier starting point. Walk the list looking for one with a source of our
-         * id.
-         */
-        WT_ERR(__curbackup_incr_blkmod(session, btree, cb));
-        /*
-         * If there is no block modification information for this file, there is no information to
-         * return to the user.
-         */
-        if (cb->bitstring.mem == NULL)
-            WT_ERR(WT_NOTFOUND);
+        if (cb->incr_init) {
```

Comment:
I think this `else { if (cb->incr_init) {X} else {Y} }` can be further simplified to `else if (cb->incr_init) {X} else {Y}`